src/common/actions/snap-builds.js

```diff
@@ -18,27 +20,38 @@ const REQUEST_POST_OPTIONS = {
   credentials: 'same-origin'
 };
 
-export function fetchBuildsSuccess(builds) {
+export function fetchBuildsSuccess(id, builds) {
   return {
     type: FETCH_BUILDS_SUCCESS,
-    payload: builds
+    payload: {
+      id,
+      builds
+    }
   };
 }
 
-export function fetchBuildsError(error) {
+export function fetchBuildsError(id, error) {
   return {
     type: FETCH_BUILDS_ERROR,
-    payload: error,
+    payload: {
+      id,
+      error
+    },
     error: true
   };
 }
 
 // Fetch snap info (self_link) for given repository
 export function fetchSnap(repositoryUrl) {
+  const { fullName } = parseGitHubRepoUrl(repositoryUrl);
+
   return (dispatch) => {
     if (repositoryUrl) {
       dispatch({
-        type: FETCH_BUILDS
+        type: FETCH_BUILDS,
+        payload: {
+          id: fullName
+        }
```

Comment:
If I understand this correctly, `FETCH_BUILDS` is called to clear any builds in the store before fetching others?

Why not have an action called `CLEAR_BUILDS` that spells out what is happening?